ldms/python/ldmsd/parser_util.py

```diff
@@ -0,0 +1,1005 @@
+                auth = check_opt('auth', self.daemons[producer['dmn_grp']][producer['daemon']]['endpoints'][ep])
+                auth_opt = check_opt('conf', self.daemons[producer['dmn_grp']][producer['daemon']]['endpoints'][ep])
+                if auth not in auth_list:
+                    auth_list[auth] = { 'conf' : auth_opt }
+                    plugin = check_opt('plugin', self.daemons[producer['dmn_grp']][producer['daemon']]['endpoints'][ep]['auth'])
+                    if plugin is None:
+                        #print(f'Please specify auth plugin type for producer "{producer["daemon"]}" with auth name "{auth}"\n'\
+                        #       'configuration file generation will continue, but auth will likely be denied.\n')
+                        plugin = auth
+                    dstr += f'auth_add name={auth} plugin={plugin}'
+                    dstr = self.write_opt_attr(dstr, 'conf', auth_list[auth]['conf'])
+            for ep in prod_group:
+                regex = False
+                producer = self.producers[group_name][ep]
+                pname = producer['name']
+                port = self.daemons[producer['dmn_grp']][producer['daemon']]['endpoints'][ep]['port']
+                xprt = self.daemons[producer['dmn_grp']][producer['daemon']]['endpoints'][ep]['xprt']
+                hostname = self.daemons[producer['dmn_grp']][producer['daemon']]['addr']
+                auth = check_opt('auth', self.daemons[producer['dmn_grp']][producer['daemon']]['endpoints'][ep])
+                ptype = producer['type']
+                reconnect = producer['reconnect']
+                dstr += f'prdcr_add name={pname} '\
+                        f'host={hostname} '\
+                        f'port={port} '\
+                        f'xprt={xprt} '\
+                        f'type={ptype} '\
+                        f'reconnect={reconnect}'
+                dstr = self.write_opt_attr(dstr, 'auth', auth)
+                last_sampler = pname
+                if 'regex' in producer:
+                    regex = True
+                    dstr += f'prdcr_start_regex regex={producer["regex"]}\n'
+                if not regex:
+                    dstr += f'prdcr_start_regex regex=.*\n'
+        return dstr, auth_list
+
+    def write_env(self, dstr, grp, dname):
+        if grp not in self.daemons:
+            return 1
+        if dname not in self.daemons[grp]:
+            return 1
+        if check_opt('environment', self.daemons[grp][dname]):
+            if type(self.daemons[grp][dname]['environment']) is not dict:
+                print(f'Error: Environment variables must be a yaml key:value dictionary\n')
+                sys.exit(22)
+            for attr in self.daemons[grp][dname]['environment']:
+                dstr += f'env {attr}={self.daemons[grp][dname]["environment"]}\n'
+        return dstr
+
+    def write_sampler(self, dstr, smplr_grp, sname):
+        if smplr_grp not in self.samplers:
```

Comment:
"self.samplers" may be None if there is not a samplers section in the yaml. This will throw an exception, so the None case needs to be properly handled.